install.sh

```diff
@@ -302,14 +326,18 @@ check_ports() {
 
 # Pull and start containers.
 deploy_containers_docker() {
+    local use_timeout=$1
+
     log_info "Pulling latest container images."
-    if ! output=$(run_with_timeout 300 docker compose pull 2>&1); then
+    if [ $use_timeout -eq 1 ] && ! output=$(run_with_timeout 300 $DOCKER_CMD compose pull 2>&1); then
         log_error "Failed to pull container images. Error: $output"
         exit 1
+    else
+        $DOCKER_CMD compose pull
```

Comment:
Should this check the return code like above and log the error and exit if the pull fails?

Reply:
Doesn't this also pull twice, if use\_timeout is 1 and the async pull succeeds on the if above?